### PR TITLE
[windows] implement checksum validation for rustup

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

implement checksum validation for rustup

#### Related issue:

https://github.com/github/c2c-actions-akvelon/issues/38

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by Alexey-Ayupov — APPROVED** [review]

File: images/win/scripts/Installers/Install-Rust.ps1 [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ################################################################################
 ##  File:  Install-Rust.ps1
 ##  Desc:  Install Rust for Windows
+##  Supply chain security: checksum validation
```
Comment (erik-bershel) [inline images/win/scripts/Installers/Install-Rust.ps1:4]: What do you think about mentioning that it is manual validation for the Rustup and controlled by package manager validation for the Rust itself?

File: images/win/scripts/Installers/Install-Rust.ps1 [hunk 2]
```diff
@@ -11,6 +12,19 @@ $env:CARGO_HOME = "C:\Users\Default\.cargo"
 # See https://rustup.rs/#
 $rustupPath = Start-DownloadWithRetry -Url "https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe" -Name "rustup-init.exe"
 
+#region Supply chain security
+Write-Verbose "Performing checksum verification"
+$local_file_hash = (Get-FileHash -Path (Join-Path ${env:TEMP} 'rustup-init.exe') -Algorithm SHA256).Hash
```
Comment (erik-bershel) [inline images/win/scripts/Installers/Install-Rust.ps1:17]: Still thinking about camelCase for the variables names.

File: images/win/scripts/Installers/Install-Rust.ps1 [hunk 3]
```diff
@@ -11,6 +12,19 @@ $env:CARGO_HOME = "C:\Users\Default\.cargo"
...
+
+$distributor_file_hash = (Invoke-RestMethod -Uri 'https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe.sha256').Trim()
+
+if ($local_file_hash -ne $distributor_file_hash) {
```
Comment (erik-bershel) [inline images/win/scripts/Installers/Install-Rust.ps1:21]: Function for hash check is already available.

File: images/win/scripts/Installers/Install-Rust.ps1 (hunk 1 above)
Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-Rust.ps1]: thank you for the suggestion, much better indeed

File: images/win/scripts/Installers/Install-Rust.ps1 (hunk 2 above)
Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-Rust.ps1]: thank you for the suggestion, much better indeed

File: images/win/scripts/Installers/Install-Rust.ps1 (hunk 3 above)
Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-Rust.ps1]: thank you for the suggestion, much better indeed

**Review by Alexey-Ayupov — APPROVED** [review]

**Pull request merged**